ERC20Votes/contracts/ERC20Votes.sol

```diff
@@ -0,0 +1,351 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+// Openzeppelin
+import "@openzeppelin/contracts/token/ERC20/extensions/draft-ERC20Permit.sol";
+import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+// Utils
+import "./util/IVotes.sol";
+import "./util/Math.sol";
+
+/**
+ * @title ERC20Votes
+ * @dev Extension of ERC20 to support Compound-like voting and delegation. This version is more generic than Compound's,
+ * and supports token supply up to 2^224^ - 1, while COMP is limited to 2^96^ - 1.
+ *
+ * NOTE: If exact COMP compatibility is required, use the {ERC20VotesComp} variant of this module.
+ *
+ * This extension keeps a history (checkpoints) of each account's vote power. Vote power can be delegated either
+ * by calling the {delegate} function directly, or by providing a signature to be used with {delegateBySig}. Voting
+ * power can be queried through the public accessors {getVotes} and {getPastVotes}.
+ *
+ * By default, token balance does not account for voting power. This makes transfers cheaper. The downside is that it
+ * requires users to delegate to themselves in order to activate checkpoints and have their voting power tracked.
+ */
+abstract contract ERC20Votes is IVotes, ERC20Permit {
```

Comment:
If this contract is the main contract for module, please remove `abstract`
https://dev.to/jamiescript/abstract-contract-and-interfaces-in-solidity-20cb#:~:text=An%20abstract%20contract%20is%20one,be%20inherited%20by%20another%20contract.
Otherwise, users in bunzz will not be able to deploy it.